### PR TITLE
sw-5519 Add Firm to Search Service

### PR DESCRIPTION
**Pull request by gurps-singh**
This PR is to add firm to the search service. This includes adding firm specific functionality but also allowing bulk operations to be completed across firms and people. Searching wise, this allows us to search across firm number, firm name, casrecnumber and the searchable field.

### CODE REVIEW

File: internal/cmd/index_test.go [hunk 1]
```diff
@@ -11,12 +11,13 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestIndex(t *testing.T) {
+// TODO flaky tests when ran locally
```
Comment (Nicholassully) [inline internal/cmd/index_test.go:14]: Is there anything we can do that will make it less flaky?

File: main.go [hunk 2]
```diff
@@ -35,32 +46,15 @@ func main() {
...
-		l.Fatal(err)
-	}
-
-	indices := []string{person.AliasName}
-	if aliasedIndex != personIndex {
-		indices = append(indices, personIndex)
-	}
-
-	l.Println("indexing to", indices)
+	//force recreate indices
```
Comment (Nicholassully) [inline main.go:55]: Same for this one too as last comment

**Review by Nicholassully — APPROVED** [review]

**Review by ramchale — DISMISSED** [review]

File: internal/cmd/commands_test.go [hunk 3]
```diff
@@ -44,21 +43,42 @@ func TestCommandsRun(t *testing.T) {
...
+	os.Args = []string{"search-service", "index person"}
 	Run(l, cmd1, cmd2)
 
-	assert.Equal(t, "what", hook.LastEntry().Message)
-	assert.Equal(t, 1, exitCode)
+	assert.Contains(t, "Running command: *cmd.MockCommand", hook.LastEntry().Message)
+	assert.Equal(t, 0, exitCode)
 }
+
+//this is the one that always failed even on main
```
[the remark this answers is not in the document]
Comment (gurps-singh) [reply internal/cmd/commands_test.go:66]: whoops, forgot that was commented out - on it

File: internal/cmd/index_test.go (hunk 1 above)
Comment (gurps-singh) [reply internal/cmd/index_test.go:14]: we couldn't find out the core reason so spun up a ticket attached to the ticket to look at this on as it passes on the second or third go locally

**Review by Nicholassully — APPROVED** [review]

**Pull request merged**